// @api3/airnode-protocol-v1/dist/src/contracts/factories/protocol/AirnodeRequester__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AirnodeRequester__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "airnodeProtocol",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var AirnodeRequester__factory = /** @class */ (function () {
    function AirnodeRequester__factory() {
    }
    AirnodeRequester__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    AirnodeRequester__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    AirnodeRequester__factory.abi = _abi;
    return AirnodeRequester__factory;
}());
exports.AirnodeRequester__factory = AirnodeRequester__factory;
