// @api3/airnode-protocol/dist/src/contracts/factories/rrp/interfaces/IAuthorizationUtilsV0__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAuthorizationUtilsV0__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address[]",
                name: "authorizers",
                type: "address[]",
            },
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "requestId",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
        ],
        name: "checkAuthorizationStatus",
        outputs: [
            {
                internalType: "bool",
                name: "status",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address[]",
                name: "authorizers",
                type: "address[]",
            },
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32[]",
                name: "requestIds",
                type: "bytes32[]",
            },
            {
                internalType: "bytes32[]",
                name: "endpointIds",
                type: "bytes32[]",
            },
            {
                internalType: "address[]",
                name: "sponsors",
                type: "address[]",
            },
            {
                internalType: "address[]",
                name: "requesters",
                type: "address[]",
            },
        ],
        name: "checkAuthorizationStatuses",
        outputs: [
            {
                internalType: "bool[]",
                name: "statuses",
                type: "bool[]",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IAuthorizationUtilsV0__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IAuthorizationUtilsV0__factory = IAuthorizationUtilsV0__factory;
IAuthorizationUtilsV0__factory.abi = _abi;
