// @api3/airnode-protocol/dist/src/contracts/factories/rrp/interfaces/ITemplateUtilsV0__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ITemplateUtilsV0__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        name: "CreatedTemplate",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        name: "createTemplate",
        outputs: [
            {
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32[]",
                name: "templateIds",
                type: "bytes32[]",
            },
        ],
        name: "getTemplates",
        outputs: [
            {
                internalType: "address[]",
                name: "airnodes",
                type: "address[]",
            },
            {
                internalType: "bytes32[]",
                name: "endpointIds",
                type: "bytes32[]",
            },
            {
                internalType: "bytes[]",
                name: "parameters",
                type: "bytes[]",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
        ],
        name: "templates",
        outputs: [
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class ITemplateUtilsV0__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ITemplateUtilsV0__factory = ITemplateUtilsV0__factory;
ITemplateUtilsV0__factory.abi = _abi;
