// @api3/airnode-protocol/dist/src/contracts/factories/rrp/interfaces/IWithdrawalUtilsV0__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IWithdrawalUtilsV0__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "address",
                name: "sponsorWallet",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
        ],
        name: "FulfilledWithdrawal",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "address",
                name: "sponsorWallet",
                type: "address",
            },
        ],
        name: "RequestedWithdrawal",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
        ],
        name: "fulfillWithdrawal",
        outputs: [],
        stateMutability: "payable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "address",
                name: "sponsorWallet",
                type: "address",
            },
        ],
        name: "requestWithdrawal",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
        ],
        name: "sponsorToWithdrawalRequestCount",
        outputs: [
            {
                internalType: "uint256",
                name: "withdrawalRequestCount",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IWithdrawalUtilsV0__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IWithdrawalUtilsV0__factory = IWithdrawalUtilsV0__factory;
IWithdrawalUtilsV0__factory.abi = _abi;
