// @api3/airnode-protocol/dist/src/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.erc721Mocks = exports.PROTOCOL_IDS = exports.references = exports.networks = exports.authorizers = exports.mocks = exports.RequesterAuthorizerWithErc721Factory = exports.RrpBeaconServerV0Factory = exports.AccessControlRegistryFactory = exports.AirnodeRrpV0DryRunFactory = exports.AirnodeRrpV0Factory = exports.RequesterAuthorizerWithAirnodeAddresses = exports.AccessControlRegistryAddresses = exports.AirnodeRrpDryRunAddresses = exports.AirnodeRrpAddresses = void 0;
const airnode_protocol_v1_1 = require("@api3/airnode-protocol-v1");
Object.defineProperty(exports, "RequesterAuthorizerWithErc721Factory", { enumerable: true, get: function () { return airnode_protocol_v1_1.RequesterAuthorizerWithErc721__factory; } });
const contracts_1 = require("./contracts");
Object.defineProperty(exports, "AirnodeRrpV0Factory", { enumerable: true, get: function () { return contracts_1.AirnodeRrpV0__factory; } });
Object.defineProperty(exports, "AirnodeRrpV0DryRunFactory", { enumerable: true, get: function () { return contracts_1.AirnodeRrpV0DryRun__factory; } });
Object.defineProperty(exports, "AccessControlRegistryFactory", { enumerable: true, get: function () { return contracts_1.AccessControlRegistry__factory; } });
Object.defineProperty(exports, "RrpBeaconServerV0Factory", { enumerable: true, get: function () { return contracts_1.RrpBeaconServerV0__factory; } });
const references_json_1 = __importDefault(require("../deployments/references.json"));
const references = {
    ...references_json_1.default,
    RequesterAuthorizerWithErc721: airnode_protocol_v1_1.references.RequesterAuthorizerWithErc721,
};
exports.references = references;
const AirnodeRrpAddresses = references.AirnodeRrpV0;
exports.AirnodeRrpAddresses = AirnodeRrpAddresses;
const AirnodeRrpDryRunAddresses = references.AirnodeRrpV0DryRun;
exports.AirnodeRrpDryRunAddresses = AirnodeRrpDryRunAddresses;
const AccessControlRegistryAddresses = references.AccessControlRegistry;
exports.AccessControlRegistryAddresses = AccessControlRegistryAddresses;
const RequesterAuthorizerWithAirnodeAddresses = references.RequesterAuthorizerWithAirnode;
exports.RequesterAuthorizerWithAirnodeAddresses = RequesterAuthorizerWithAirnodeAddresses;
const networks = references.networks;
exports.networks = networks;
const PROTOCOL_IDS = {
    RRP: '1',
    PSP: '2',
    RELAYED_RRP: '3',
    RELAYED_PSP: '4',
    AIRSEEKER: '5',
    AIRKEEPER: '12345',
};
exports.PROTOCOL_IDS = PROTOCOL_IDS;
// TODO: This and the mock exports below should be defined in airnode-operations instead and not exported from here
const erc721Mocks = {
    MockErc721Factory: airnode_protocol_v1_1.MockErc721__factory,
};
exports.erc721Mocks = erc721Mocks;
const mocks = {
    MockRrpRequesterFactory: contracts_1.MockRrpRequesterV0__factory,
};
exports.mocks = mocks;
const authorizers = {
    RequesterAuthorizerWithAirnodeFactory: contracts_1.RequesterAuthorizerWithAirnode__factory,
};
exports.authorizers = authorizers;
