// @api3/airnode-protocol-v1/dist/src/contracts/factories/utils/mock/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MockSelfMulticall__factory = exports.MockMulticallTarget__factory = exports.MockExternalMulticall__factory = exports.MockExpiringMetaTxForwarderTarget__factory = exports.MockCallForwarderTarget__factory = exports.mockErc20PermitTokenSol = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.mockErc20PermitTokenSol = __importStar(require("./MockErc20PermitToken.sol"));
var MockCallForwarderTarget__factory_1 = require("./MockCallForwarderTarget__factory");
Object.defineProperty(exports, "MockCallForwarderTarget__factory", { enumerable: true, get: function () { return MockCallForwarderTarget__factory_1.MockCallForwarderTarget__factory; } });
var MockExpiringMetaTxForwarderTarget__factory_1 = require("./MockExpiringMetaTxForwarderTarget__factory");
Object.defineProperty(exports, "MockExpiringMetaTxForwarderTarget__factory", { enumerable: true, get: function () { return MockExpiringMetaTxForwarderTarget__factory_1.MockExpiringMetaTxForwarderTarget__factory; } });
var MockExternalMulticall__factory_1 = require("./MockExternalMulticall__factory");
Object.defineProperty(exports, "MockExternalMulticall__factory", { enumerable: true, get: function () { return MockExternalMulticall__factory_1.MockExternalMulticall__factory; } });
var MockMulticallTarget__factory_1 = require("./MockMulticallTarget__factory");
Object.defineProperty(exports, "MockMulticallTarget__factory", { enumerable: true, get: function () { return MockMulticallTarget__factory_1.MockMulticallTarget__factory; } });
var MockSelfMulticall__factory_1 = require("./MockSelfMulticall__factory");
Object.defineProperty(exports, "MockSelfMulticall__factory", { enumerable: true, get: function () { return MockSelfMulticall__factory_1.MockSelfMulticall__factory; } });
