// @api3/airnode-protocol-v1/dist/src/contracts/factories/api3-server-v1/proxies/interfaces/IDapiProxy__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IDapiProxy__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "api3ServerV1",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "dapiNameHash",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "read",
        outputs: [
            {
                internalType: "int224",
                name: "value",
                type: "int224",
            },
            {
                internalType: "uint32",
                name: "timestamp",
                type: "uint32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IDapiProxy__factory = /** @class */ (function () {
    function IDapiProxy__factory() {
    }
    IDapiProxy__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IDapiProxy__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IDapiProxy__factory.abi = _abi;
    return IDapiProxy__factory;
}());
exports.IDapiProxy__factory = IDapiProxy__factory;
