// @api3/airnode-protocol-v1/dist/src/contracts/factories/api3-server-v1/proxies/interfaces/IOevProxy__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOevProxy__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "oevBeneficiary",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IOevProxy__factory = /** @class */ (function () {
    function IOevProxy__factory() {
    }
    IOevProxy__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IOevProxy__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IOevProxy__factory.abi = _abi;
    return IOevProxy__factory;
}());
exports.IOevProxy__factory = IOevProxy__factory;
