// @api3/airnode-protocol-v1/dist/src/contracts/factories/protocol/interfaces/IStorageUtils__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IStorageUtils__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "subscriptionId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "chainId",
                type: "uint256",
            },
            {
                indexed: false,
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "endpointOrTemplateId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
            {
                indexed: false,
                internalType: "bytes",
                name: "conditions",
                type: "bytes",
            },
            {
                indexed: false,
                internalType: "address",
                name: "relayer",
                type: "address",
            },
            {
                indexed: false,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: false,
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "fulfillFunctionId",
                type: "bytes4",
            },
        ],
        name: "StoredSubscription",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        name: "StoredTemplate",
        type: "event",
    },
    {
        inputs: [],
        name: "MAXIMUM_PARAMETER_LENGTH",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "chainId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "endpointOrTemplateId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
            {
                internalType: "bytes",
                name: "conditions",
                type: "bytes",
            },
            {
                internalType: "address",
                name: "relayer",
                type: "address",
            },
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "fulfillFunctionId",
                type: "bytes4",
            },
        ],
        name: "storeSubscription",
        outputs: [
            {
                internalType: "bytes32",
                name: "subscriptionId",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        name: "storeTemplate",
        outputs: [
            {
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "subscriptionId",
                type: "bytes32",
            },
        ],
        name: "subscriptions",
        outputs: [
            {
                internalType: "uint256",
                name: "chainId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "airnode",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "endpointOrTemplateId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
            {
                internalType: "bytes",
                name: "conditions",
                type: "bytes",
            },
            {
                internalType: "address",
                name: "relayer",
                type: "address",
            },
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                internalType: "address",
                name: "requester",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "fulfillFunctionId",
                type: "bytes4",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "templateId",
                type: "bytes32",
            },
        ],
        name: "templates",
        outputs: [
            {
                internalType: "bytes32",
                name: "endpointId",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "parameters",
                type: "bytes",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IStorageUtils__factory = /** @class */ (function () {
    function IStorageUtils__factory() {
    }
    IStorageUtils__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IStorageUtils__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IStorageUtils__factory.abi = _abi;
    return IStorageUtils__factory;
}());
exports.IStorageUtils__factory = IStorageUtils__factory;
