// @api3/airnode-protocol-v1/dist/src/contracts/factories/protocol/interfaces/IWithdrawalUtils__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IWithdrawalUtils__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
        ],
        name: "ClaimedBalance",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "airnodeOrRelayer",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "protocolId",
                type: "uint256",
            },
            {
                indexed: false,
                internalType: "address",
                name: "sponsorWallet",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "amount",
                type: "uint256",
            },
        ],
        name: "FulfilledWithdrawal",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "airnodeOrRelayer",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                indexed: true,
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "protocolId",
                type: "uint256",
            },
        ],
        name: "RequestedWithdrawal",
        type: "event",
    },
    {
        inputs: [],
        name: "claimBalance",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
            {
                internalType: "address",
                name: "airnodeOrRelayer",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "protocolId",
                type: "uint256",
            },
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "timestamp",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "signature",
                type: "bytes",
            },
        ],
        name: "fulfillWithdrawal",
        outputs: [],
        stateMutability: "payable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "airnodeOrRelayer",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "protocolId",
                type: "uint256",
            },
        ],
        name: "requestWithdrawal",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
        ],
        name: "sponsorToBalance",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "sponsor",
                type: "address",
            },
        ],
        name: "sponsorToWithdrawalRequestCount",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "withdrawalRequestId",
                type: "bytes32",
            },
        ],
        name: "withdrawalRequestIsAwaitingFulfillment",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IWithdrawalUtils__factory = /** @class */ (function () {
    function IWithdrawalUtils__factory() {
    }
    IWithdrawalUtils__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IWithdrawalUtils__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IWithdrawalUtils__factory.abi = _abi;
    return IWithdrawalUtils__factory;
}());
exports.IWithdrawalUtils__factory = IWithdrawalUtils__factory;
