// @api3/airnode-protocol-v1/dist/src/contracts/factories/utils/interfaces/IExternalMulticall__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IExternalMulticall__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address[]",
                name: "targets",
                type: "address[]",
            },
            {
                internalType: "bytes[]",
                name: "data",
                type: "bytes[]",
            },
        ],
        name: "externalMulticall",
        outputs: [
            {
                internalType: "bytes[]",
                name: "returndata",
                type: "bytes[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address[]",
                name: "targets",
                type: "address[]",
            },
            {
                internalType: "bytes[]",
                name: "data",
                type: "bytes[]",
            },
        ],
        name: "tryExternalMulticall",
        outputs: [
            {
                internalType: "bool[]",
                name: "success",
                type: "bool[]",
            },
            {
                internalType: "bytes[]",
                name: "returndata",
                type: "bytes[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var IExternalMulticall__factory = /** @class */ (function () {
    function IExternalMulticall__factory() {
    }
    IExternalMulticall__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IExternalMulticall__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IExternalMulticall__factory.abi = _abi;
    return IExternalMulticall__factory;
}());
exports.IExternalMulticall__factory = IExternalMulticall__factory;
