// @api3/airnode-protocol-v1/dist/src/contracts/factories/utils/interfaces/IOevSearcherMulticallV1__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOevSearcherMulticallV1__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address[]",
                name: "targets",
                type: "address[]",
            },
            {
                internalType: "bytes[]",
                name: "data",
                type: "bytes[]",
            },
            {
                internalType: "uint256[]",
                name: "values",
                type: "uint256[]",
            },
        ],
        name: "externalMulticallWithValue",
        outputs: [
            {
                internalType: "bytes[]",
                name: "returndata",
                type: "bytes[]",
            },
        ],
        stateMutability: "payable",
        type: "function",
    },
];
var IOevSearcherMulticallV1__factory = /** @class */ (function () {
    function IOevSearcherMulticallV1__factory() {
    }
    IOevSearcherMulticallV1__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IOevSearcherMulticallV1__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IOevSearcherMulticallV1__factory.abi = _abi;
    return IOevSearcherMulticallV1__factory;
}());
exports.IOevSearcherMulticallV1__factory = IOevSearcherMulticallV1__factory;
