// @api3/airnode-protocol-v1/dist/src/contracts/factories/utils/interfaces/IOwnableCallForwarder__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOwnableCallForwarder__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "forwardTarget",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "forwardedCalldata",
                type: "bytes",
            },
        ],
        name: "forwardCall",
        outputs: [
            {
                internalType: "bytes",
                name: "returnedData",
                type: "bytes",
            },
        ],
        stateMutability: "payable",
        type: "function",
    },
];
var IOwnableCallForwarder__factory = /** @class */ (function () {
    function IOwnableCallForwarder__factory() {
    }
    IOwnableCallForwarder__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IOwnableCallForwarder__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IOwnableCallForwarder__factory.abi = _abi;
    return IOwnableCallForwarder__factory;
}());
exports.IOwnableCallForwarder__factory = IOwnableCallForwarder__factory;
