// @api3/airnode-protocol-v1/dist/src/contracts/factories/utils/interfaces/ISelfMulticall__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISelfMulticall__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "bytes[]",
                name: "data",
                type: "bytes[]",
            },
        ],
        name: "multicall",
        outputs: [
            {
                internalType: "bytes[]",
                name: "returndata",
                type: "bytes[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes[]",
                name: "data",
                type: "bytes[]",
            },
        ],
        name: "tryMulticall",
        outputs: [
            {
                internalType: "bool[]",
                name: "successes",
                type: "bool[]",
            },
            {
                internalType: "bytes[]",
                name: "returndata",
                type: "bytes[]",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var ISelfMulticall__factory = /** @class */ (function () {
    function ISelfMulticall__factory() {
    }
    ISelfMulticall__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ISelfMulticall__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ISelfMulticall__factory.abi = _abi;
    return ISelfMulticall__factory;
}());
exports.ISelfMulticall__factory = ISelfMulticall__factory;
